workers/network-manager/src/executor/pubsub.rs

```diff
@@ -43,14 +43,18 @@ impl NetworkManagerExecutor {
     }
 
     /// Handle an incoming network request for a pubsub message
-    pub(super) fn handle_inbound_pubsub_message(
+    pub(super) async fn handle_inbound_pubsub_message(
         &self,
         message: GossipsubMessage,
     ) -> Result<(), NetworkManagerError> {
         // Deserialize into API types and verify auth
+        let pkey = self.cluster_key.public;
         let event: AuthenticatedPubsubMessage =
             message.data.try_into().map_err(NetworkManagerError::Serialization)?;
-        event.verify_cluster_auth(&self.cluster_key.public)?;
+        let event =
```

Comment:
tiny nit but can we drop a small comment here regarding _why_ we spawn on the blocking pool? is just helpful for an uncontextualized reader